File: app/src/main/java/com/example/rodrigo/guessinggame/GameBoardContract.java
package com.example.rodrigo.guessinggame;

/**
 * Created by rodrigo on 25/11/15.
 */
interface GameBoardContract {

    interface View {
        void showWelcomeMessage(String text);

        void showQuestion(String text);

        void finishGame(String text);

        void showAddNewAnimal(String text);

        void showAddNewAnimalQuestion(String text);
    }

    interface UserActionsListener {
        void newGame();

        void startGame();

        void answer(boolean answer);

        void addAnimal(String name);

        void addQuestion(String text);
    }
}

File: app/src/main/java/com/example/rodrigo/guessinggame/GameBoardContract.kt
package com.example.rodrigo.guessinggame

interface GameBoardContract {
    interface View {
        fun showWelcomeMessage(text: String?)
        fun showQuestion(text: String?)
        fun finishGame(text: String?)
        fun showAddNewAnimal(text: String?)
        fun showAddNewAnimalQuestion(text: String?)
    }

    interface UserActionsListener {
        fun newGame()
        fun startGame()
        fun answer(answer: Boolean)
        fun addAnimal(name: String?)
        fun addQuestion(text: String?)
    }
}